index.tsx

```diff
@@ -238,85 +236,111 @@ function Folder({
   }, [content])
 
   return (
-    <NavigationStack>
-      <List
-        navigationTitle={Path.basename(path)}
-        toolbar={{
-          confirmationAction:
-            select === "folder" ? (
-              <Button
-                title={"Select here"}
-                action={() => onSelection(path, dismiss)}
-              />
-            ) : (
-              <EmptyView />
-            ),
-          cancellationAction: (
-            <Button
-              title={"Cancel"}
-              action={() => cancel(dismiss)}
-            />
-          ),
-          topBarTrailing: (
+    <List
+      navigationTitle={Path.basename(path)}
+      toolbar={{
+        bottomBar: select === "folder" && !topLevel ?
+          <HStack>
+            <Spacer />
             <Button
-              systemImage="plus"
-              title={""}
-              action={async () => {
-                const name = await Dialog.prompt({
-                  title: "New Folder",
-                  cancelLabel: "Cancel",
-                  confirmLabel: "Create",
-                })
-                if (name) {
-                  const folder = Path.join(path, name)
-                  FileManager.createDirectorySync(folder)
-                  setContent([folder, ...content])
-                }
+              action={() => {
+                dismiss()
```

Comment:
When selecting a file more than one folder deep it also dismisses the main view. Although this might never be needed, it would be nice to fix it now. You never know.

Selecting a file just one folder deep works.